web3/_utils/module_testing/utils.py

```diff
@@ -113,12 +110,11 @@ def __init__(
                 "AsyncMakeRequestFn", "MakeRequestFn"
             ] = w3.provider.make_request
 
+        self._mock_request_counter = 1
+
     def _build_request_id(self) -> int:
-        request_id = (
-            next(copy.deepcopy(self.w3.provider.request_counter))
```

Comment:
This was raising a: `TypeError: cannot pickle 'itertools.count'` because itertools objects can no longer be deepcopied or pickled in 3.14. So I opted to just manually track it here. The logic in the WebSocket Providers remains the same.

Reply:
Yeah I think that works well. There's an issue to track this [here](https://github.com/ethereum/web3.py/issues/3511). I think the fact that we just use our own counter shouldn't interfere with anything right? And I think what was mentioned in the `PersistentConnectionProvider` was fixed a while ago. So I think both that issue and my stale PR can be closed :)